### PR TITLE
Docker fixes (related to #3695)

### PR DESCRIPTION
**Pull request by filimonov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Check https://github.com/yandex/ClickHouse/pull/3695#issuecomment-445360846 and further comments
1) waiting for server start (via http pings)
2) clickhouse-client added to clickhouse-server image, that is just a symlink but makes it possible to connect to the server running in the container in a natural way with simple `sudo docker container exec -it container_name clickhouse-client`. Related to #580
3) wget added (actually waiting for accepting connections can be implemented w/o it, but wget is quite [tiny](https://packages.ubuntu.com/xenial/wget) and handy)
4) greps changed to `clickhouse extract-from-config`
5) format_schema_path added
6) gosu moved higher (that makes rebuilds of image connected to lines below it cheaper)

### CODE REVIEW

File: docker/server/Dockerfile [hunk 1]
```diff
@@ -16,19 +16,29 @@ RUN apt-get update \
     && env DEBIAN_FRONTEND=noninteractive \
         apt-get install --allow-unauthenticated --yes --no-install-recommends \
             clickhouse-common-static=$version \
+            clickhouse-client=$version \
             clickhouse-server=$version \
             libgcc-7-dev \
```
Comment (filimonov) [inline docker/server/Dockerfile]: why is it needed?

Comment (alexey-milovidov) [reply docker/server/Dockerfile]: For runtime code generation.
BTW, we only need "startup files": `crti.o`, etc.

Comment (alexey-milovidov) [reply docker/server/Dockerfile]: Better to provide these files in package.

**Comment by alexey-milovidov** [comment]
> waiting for server start (via http pings)

The only concern is that the user may disable HTTP completely (they can leave HTTPS only).
(For example, this is the case for Yandex.Cloud).

**Comment by alexey-milovidov** [comment]
> greps changed to clickhouse extract-from-config
format_schema_path added

Great! Thank you!

**Comment by filimonov** [comment]
> The only concern is that the user may disable HTTP completely (they can leave HTTPS only).
(For example, this is the case for Yandex.Cloud).

Actually, every separate part can be disabled. `tcp_port` can be disabled too. Secure ports are not enabled by default. Message in log can be suppressed by log level. So at that point there is no any guaranteed way that clickhouse is working. 

I've checked if it's possible to postpone daemonization of the process until finishing of initialization, but it looks like it will require changing default behaviour of poco's ServerApplication. Not sure, may be it  can be done with using poco's [Subsystem](https://pocoproject.org/docs/Poco.Util.Subsystem.html).

**Pull request merged**